src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,45 @@
+package duke.command;
+
+import duke.Storage;
+import duke.exception.DukeException;
+import duke.task.TaskList;
+
+/**
+ * Abstract command class that can be inherited by various commands
+ */
+public abstract class Command {
+    protected String commandType;
+    protected String commandDetails;
```

Comment:
Perhaps this could be commandDetail instead as it might not need to be represented in its plural form as it is not a list or array of objects.